### PR TITLE
bpo-36071 Add support for Windows ARM32 in ctypes/libffi

### PR DESCRIPTION
**Pull request by paulmon**
test_SEH still fails.  I'm trying to track down unwind semantics for armasm.exe if possible.

### CODE REVIEW

**Comment by paulmon** [comment]
@zooba @zware  
These ARM changes include unwind semantics now, and have also been submitted to libffi.
https://github.com/libffi/libffi/pull/477

File: msvcc.sh [hunk 1]
```diff
@@ -80,6 +80,11 @@ do
       safeseh=
       shift 1
     ;;
+    -marm)
```
Comment (paulmon) [inline msvcc.sh:83]: This is where -marm is defined.

**Comment by zooba** [comment]
Okay, I didn't notice this patch, sorry.

We're diverging significantly from libffi now, and I'm not so keen on that just yet. How can we help move along the patch on their side?

**Comment by paulmon** [comment]
I pinged the PR discussion and sent email to the libffi-discussion alias.
https://github.com/libffi/libffi/pull/477
https://sourceware.org/ml/libffi-discuss/2019/msg00003.html

**Comment by zooba** [comment]
I'm going to merge this, but @paulmon if it doesn't make it into libffi then we'll need the patch again next time we take an update from them.

**Comment by paulmon** [comment]
Understood. Thanks!

**Comment by paulmon** [comment]
These changes were merged into libffi/master on 4/25
https://github.com/libffi/libffi/pull/477#event-2302888066

**Pull request merged**